edx/xblock-lti-consumer
# [BD-24][TNL-7607] BB-3072: Deep Linking response endpoint and model implementation

This PR implements the Deep Linking response endpoint, which tools use to send back information to the LMS after the user selects the content and configuration in the Tool's UI. 

Split from #112 

**Testing instructions:**
__Incomplete, tool configuration instructions TBD__
1. Checkout this branch.
2. Set up a local LTI tool (recommended: https://github.com/dmitry-viskov/pylti1.3-django-example) and ngrok tunnel.
3. Apply this patch to allow Deep Link launches:
```
diff --git a/lti_consumer/lti_xblock.py b/lti_consumer/lti_xblock.py
index df0cf16..3e5cadd 100644
--- a/lti_consumer/lti_xblock.py
+++ b/lti_consumer/lti_xblock.py
@@ -1378,7 +1378,7 @@ class LtiConsumerXBlock(StudioEditableXBlockMixin, XBlock):
             # Retrieve and set LTI 1.3 Launch start URL
             lti_block_launch_handler = get_lti_1p3_launch_start_url(
                 block=self,
-                deep_link_launch=False,
+                deep_link_launch=True,
                 hint=str(self.location)  # pylint: disable=no-member
             )
```
4. Do a LTI launch (from the LMS).
5. Select an option in the LTI tool's UI.
6. Go to http://localhost:18000/admin/lti_consumer/ltidlcontentitem/ and check that the option you selected is stored in the model. Make note of the ID.
7. Do the launch again, and check that the old option was deleted and only the new one is stored.

**Author notes:**
This does not implement content presentation, just stores the data the user selected in the models.
